specs/agents/configuration.md

```diff
@@ -10,6 +10,31 @@ Here's a list of the config options across all agents, their types, default valu
 
 They are provided as environment variables but depending on the language there might be several feasible ways to let the user tweak them. For example besides the environment variable `ELASTIC_APM_SERVER_URL`, the Node.js Agent might also allow the user to configure the server URL via a config option named `serverUrl`, while the Python Agent might also allow the user to configure it via a config option named `server_url`.
 
+### Configuration Value Types
+
+The following table enumerates the available configuration types across the
+agents:
+
+
+| Type | Description (if needed) |
+|------|-------------------------|
+| String   |  |
+| Integer  |  |
+| Float    |  |
+| Boolean  | Encoded as a lower-case boolean string: `"false"`, `"true"` |
+| List     | Encoded as a comma-separated string (whitespace surrounding items should be stripped): `"foo,bar,baz"` |
+| Mapping  | Encoded as a string, with `"key=value"` pairs separated by commas (whitespace surrounding items should be stripped): `"foo=bar,baz=foo"` |
+| Duration | String with duration encoded using unit suffixes (`ms` for millisecond, `s` for second, `m` for minute). |
+
+#### Duration Config Legacy Considerations
+
+For duration-formatted config options, some agents allow users to omit the unit
+suffix for backwards compatibility reasons. Going forward, all
+duration-formatted config options should require the unit suffix, falling back
+to the default value if an invalid value is provided.  Existing
+duration-formatted config options should be changed to require the unit suffix
+at the next major version.
+
```

Comment:
```
### Invalid Configuration Values

If an invalid value for a configuration option is provided (for example: `breakdown_metrics="yes"`, `apiRequestTime="1h"`) then the agent MUST ignore the value (falling back to a higher scope or default value) and SHOULD emit a log warning about the ignored value.
```

@basepi Felix [asked me](https://github.com/elastic/apm-agent-nodejs/issues/2120#issuecomment-866109084) to include a note about handling invalid config values. Would it make sense to include something like the above to your PR? I'm happy to do a separate PR after yours is merged, if you prefer.

Reply:
I'll add it, no problem.